src/elements/common/annotator-context/withAnnotations.tsx

```diff
@@ -75,12 +74,18 @@ export default function withAnnotations<P extends object>(
             }
         };
 
-        handleOnAnnotator = (annotator: Annotator): void => {
+        handleAnnotator = (annotator: Annotator): void => {
             this.annotator = annotator;
+            this.annotator.addListener('annotatorevent', this.handleAnnotatorEvent);
         };
 
         handlePreviewDestroy = (): void => {
             this.setState(defaultState);
+
+            if (this.annotator) {
+                this.annotator.removeAllListeners();
```

Comment:
We should avoid using `removeAllListeners` in nearly all circumstances, since it can cause side effects for other components. We know what listeners we added, so we can remove them explicitly.